openhands/controller/agent_controller.py

```diff
@@ -449,7 +449,21 @@ async def _step(self) -> None:
                 == ActionConfirmationStatus.AWAITING_CONFIRMATION
             ):
                 await self.set_agent_state_to(AgentState.AWAITING_USER_CONFIRMATION)
-            self.event_stream.add_event(action, EventSource.AGENT)
+
+            if isinstance(action, CmdRunAction) and '\n' in action.command:
+                # Split the command into multiple CmdRunAction instances
+                commands = action.command.split('\n')
```

Comment:
So we can't just split on `\n`--bash commands are way more complicated than that.

We have some existing logic here to do this better: https://github.com/All-Hands-AI/OpenHands/blob/feee509de7ffe30859611ace9882c72d1a6a3a6b/openhands/runtime/utils/bash.py#L6

Reply:
Replaced logic.